analyze/analyses/utils.rs

```diff
@@ -0,0 +1,14 @@
+use std::collections::BTreeSet;
+
+use petgraph::visit::Walker;
+
+use twiggy_ir as ir;
+
+pub fn get_unreachable_items<'a>(items: &'a ir::Items) -> impl Iterator<Item = &'a ir::Item> {
```

Comment:
This logic is the same that was used in the garbage subcommand, but has been outlined into a `utils` module, so that we can call it in the dominators analysis too. I am open to feedback on where this should be placed, or what this submodule should be called, since I am always hesitant about `utils `modules that can quickly become catch-all junk drawers :T

Reply:
Yeah, I'm always hesitant as well :) In this case, I think we could leave this in the `garbage` module, but make it `pub(crate)` so that teh dominators analysis can use it.